packages/jasmine/src/jasmine_runner.js

```diff
@@ -51,24 +51,35 @@ Error.stackTraceLimit = Infinity;
 const IS_TEST_FILE = /[^a-zA-Z0-9](spec|test)\.js$/i;
 const IS_NODE_MODULE = /\/node_modules\//
 
+// We process arguments by splicing them out of the process.argv
+// Users could set their own templated_args on their test, then
+// the tested code might process the argv
+// So it shouldn't see these Bazel-specific ones
+function readArg() {
+  return process.argv.splice(2, 1)[0];
+}
+
 function main(args) {
-  if (!args.length) {
-    throw new Error('Spec file manifest expected argument missing');
+  if (args.length < 3) {
+    throw new Error('expected argument missing');
   }
+
+
   // first args is always the path to the manifest
-  const manifest = require.resolve(args[0]);
+  const manifest = require.resolve(readArg());
   // second is always a flag to enable coverage or not
-  const coverageArg = args[1];
+  const coverageArg = readArg();
   const enableCoverage = coverageArg === '--coverage';
-
-  // Remove the manifest, some tested code may process the argv.
-  // Also remove the --coverage flag
-  process.argv.splice(2, 2)[0];
+  // config file is the next arg
+  const configFile = readArg();
 
   // the relative directory the coverage reporter uses to find anf filter the files
-  const cwd = process.cwd()
+  const cwd = process.cwd();
 
   const jrunner = new JasmineRunner({jasmineCore: jasmineCore});
+  if (configFile !== '--noconfig') {
+    jrunner.loadConfigFile(require.resolve(configFile));
+  }
```

Comment:
sourceFiles will be incorrect for coverage when enabled as it only considers `IS_TEST_FILE`. Maybe leave a TODO for that although I don't see an easy way to fix it?

I've been thinking for a while about adding a `specs` and `non_specs` attributes to jasmine_node_test (as well as protractor & karma since they have the same spec filtering issue (https://github.com/bazelbuild/rules_nodejs/issues/513#issuecomment-478135251). This would keep compat and mean that `srcs` is still just strictly source files (as per bazel idiom) and `srcs` and `deps` would still get the automatic filter applies. For users that want fine grained control over the sources and specs then could switch to `specs` and `non_specs` (not sure about naming) which would accept source files and handle the javascript provider.

The above would also mean changing jasmine_node_test to a proper rule and removing the macro which is just a wrapper around nodejs_test.

Reply:
I added more notes to the docs for the new attribute and marked it experimental.

Yeah the fact that it's just a macro made this change harder too, but it didn't feel like the right time to dive into refactoring...